src/clj/witan/send/send.clj

```diff
@@ -234,26 +293,25 @@
-        states-to-change (if (= 1 modify-transition-by)
-                           nil
-                           (let [state-pairs (->> settings-to-change
-                                                  ds/row-maps
-                                                  (map #(vector (:setting-1 %) (:setting-2 %))))]
-                             (vec (mapcat (fn [year]
-                                            (mapcat (fn [age]
-                                                      (mapcat (fn [need]
-                                                                (map (fn [setting] (vector (generate-transition-key age year need (first setting))
-                                                                                           (generate-transition-key age year need (second setting))))
-                                                                     state-pairs)) valid-needs)) ages)) years))))
-        transition-matrix (if (= 1 modify-transition-by)
-                            (ds/row-maps transition-matrix)
-                            (let [convert (-> transition-matrix
-                                              ds/row-maps
-                                              u/full-transitions-map)
-                                  result (reduce (fn [m k] (modify-transitions m k * modify-transition-by)) convert states-to-change)]
-                              (mapcat (fn [[k v]] (u/back-to-transitions-matrix k v)) result)))
-        transitions (u/transitions-map transition-matrix)
-        transition-matrix-filtered (filter #(= (:calendar-year %) 2016) transition-matrix)
+        _ (when (not= 1 modify-transition-by)
+            (prn (str "Modifying transitions by " modify-transition-by)))
+        states-to-change (when (not= 1 modify-transition-by)
+                           (build-states-to-change settings-to-change valid-needs ages years))
+        transition-matrix (ds/row-maps transition-matrix)
+        modified-transition-matrix (when (not= 1 modify-transition-by)
+                                     (let [convert (-> transition-matrix
+                                                       u/full-transitions-map)
+                                           result (reduce (fn [m k] (modify-transitions m k * modify-transition-by)) convert states-to-change)]
+                                       (mapcat (fn [[k v]] (u/back-to-transitions-matrix k v)) result)))
+        _ (if (nil? modified-transition-matrix)
+            (prn "Using input transitions matrix")
+            (prn "Using modified transitions matrix"))
+        transitions (if (nil? modified-transition-matrix)
+                      (u/transitions-map transition-matrix)
+                      (u/transitions-map modified-transition-matrix))
+        transition-matrix-filtered (if (nil? modified-transition-matrix)
+                                     (filter #(= (:calendar-year %) 2016) transition-matrix)
```

Comment:
would make 2016 a constant with an expressive name (def), likely to change in the future

Reply:
😄 Please see my next PR

Reply:
The next PR makes the `2016` an input parameter to the model